### PR TITLE
Add a singleton IHttpRuntime to abstract runtime data

### PR DESCRIPTION
**Pull request by twsouthwick**
Currently, anything that relies on HttpRuntime ends up relying on static data that is initialized differently depending on if you're on IIS or not. This change introduces an abstraction so things that require HttpRuntime can build on it in a testable way.

### CODE REVIEW

File: samples/ClassLibrary/RequestInfo.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ public static void WriteRequestInfo(bool suppress)
         using (var writer = new SimpleJsonWriter(context.Response))
         {
             writer.Write("VirtualDirectory", HttpRuntime.AppDomainAppVirtualPath);
+            writer.Write("VirtualDirectory", context.Request.ApplicationPath);
```
Comment (mjrousos) [inline samples/ClassLibrary/RequestInfo.cs:16]: Nit-pick - same name as the above line could be a little confusing

File: src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs [hunk 2]
```diff
@@ -17,6 +16,7 @@ public static class SystemWebAdaptersExtensions
     public static ISystemWebAdapterBuilder AddSystemWebAdapters(this IServiceCollection services)
     {
         services.AddHttpContextAccessor();
+        services.AddSingleton<IHttpRuntime>(_ => HttpRuntimeFactory.Create());
```
Comment (mjrousos) [inline src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs:19]:
```suggestion
        services.AddSingleton<IHttpRuntime>(HttpRuntimeFactory.Create);
```

**Review by mjrousos — APPROVED** [review]

File: src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs (hunk 2 above)
Comment (twsouthwick) [reply src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs:19]: That wouldn't work - there's no overload that takes a `Func<T>` but rather it expects a `Func<IServiceProvider,T>`

Comment (mjrousos) [reply src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/SystemWebAdaptersExtensions.cs:19]: Ah, good point. I missed that it expects a service provider.

**Review by Tratcher — APPROVED** [review]

**Review by HaoK — APPROVED** [review]

**Pull request merged**